strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetArtifactController.java

```diff
@@ -661,56 +666,54 @@ private void writePackagePart(String boundaryString,
     private URI storePackage(String storageId,
                              String repositoryId,
                              InputStream is)
-        throws Exception
-    {
-        try (TempNupkgFile nupkgFile = new TempNupkgFile(is))
-        {
-            if (nupkgFile.getNuspecFile() == null)
-            {
-                return null;
-            }
-
-            String path = String.format("%s/%s/%s.%s.nupkg",
-                                        nupkgFile.getId(),
-                                        nupkgFile.getVersion(),
-                                        nupkgFile.getId(),
-                                        nupkgFile.getVersion());
-
-            nugetArtifactManagementService.validateAndStore(storageId, repositoryId, path, nupkgFile.getStream());
+            throws Exception {
 
-            Path nuspecFile = Files.createTempFile(nupkgFile.getId(), "nuspec");
-            try (OutputStream outputStream = new BufferedOutputStream(Files.newOutputStream(nuspecFile)))
-            {
-                nupkgFile.getNuspecFile().saveTo(outputStream);
-            }
-            path = String.format("%s/%s/%s.nuspec", nupkgFile.getId(), nupkgFile.getVersion(), nupkgFile.getId());
+        String path;
+        NuspecFile nuspecFile;
+        byte[] byteArray = IOUtils.toByteArray(is);
```

Comment:
@gokhankuyucak I think we shouldn't store potentially large binary contents in memory as it can cause OutOfMemory issues. Also can you please check similar issues against all PR changes.

Reply:
Is there any example to check like maven or npm package read from file? I tried to use the same InputStream for both nupkg and nuspec file read, I failed, because when I started to use for one them, the index location is changing and it's getting useless for other file read process.

Reply:
you should write HTTP stream into tmp file and then read it as much as you need with new InputStream instances from File System. It was implemented like this before.